### PR TITLE
CLN: fix asv setup_cache() for groupby.AggFunctions benchmarks

### PR DESCRIPTION
**Pull request by qwhelan**
This PR fixes a typo in an `asv` benchmark and additionally clarifies a misleading result.

Simply fixing the typo leads to this rather surprising result where the single column benchmark is ~6x slower than the multi-column benchmark:
```
$ asv dev -b groupby.AggFunctions
· Discovering benchmarks
· Running 4 total benchmarks (1 commits * 1 environments * 4 benchmarks)
[  0.00%] ·· Benchmarking existing-py_home_chris_anaconda3_bin_python
[ 12.50%] ··· Setting up groupby.py:213                                                                                                                                   ok
[ 12.50%] ··· groupby.AggFunctions.time_different_numpy_functions                                                                                                   53.7±0ms
[ 25.00%] ··· groupby.AggFunctions.time_different_python_functions_multicol                                                                                         84.7±0ms
[ 37.50%] ··· groupby.AggFunctions.time_different_python_functions_singlecol                                                                                         483±0ms
[ 50.00%] ··· groupby.AggFunctions.time_different_str_functions                                                                                                     46.0±0ms
```
Fortunately, the difference is solely due to the inclusion of a `str` column in the single column case. Therefore, we drop that column to keep the comparison comparable:
```
$ asv run HEAD -b groupby.AggFunctions
· Creating environments
· Discovering benchmarks..
·· Uninstalling from conda-py3.6-Cython-matplotlib-numexpr-numpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt...
·· Building d3fa9d67 <groupby_agg_functions> for conda-py3.6-Cython-matplotlib-numexpr-numpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt.............................
·· Installing d3fa9d67 <groupby_agg_functions> into conda-py3.6-Cython-matplotlib-numexpr-numpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt....
· Running 53900 total benchmarks (13475 commits * 1 environments * 4 benchmarks)
[  0.00%] · For pandas commit d3fa9d67 <groupby_agg_functions>:
[  0.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numexpr-numpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[  0.00%] ··· Setting up groupby.py:213                                                                                                                                   ok
[  0.00%] ··· Running (groupby.AggFunctions.time_different_numpy_functions--)....
[  0.00%] ··· groupby.AggFunctions.time_different_numpy_functions                                                                                                   23.2±3ms
[  0.01%] ··· groupby.AggFunctions.time_different_python_functions_multicol                                                                                        70.2±10ms
[  0.01%] ··· groupby.AggFunctions.time_different_python_functions_singlecol                                                                                       38.5±10ms
[  0.01%] ··· groupby.AggFunctions.time_different_str_functions                                                                                                   22.4±0.3ms
```
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: asv_bench/benchmarks/groupby.py [hunk 1]
```diff
@@ -235,7 +235,7 @@ def time_different_python_functions_multicol(self, df):
         df.groupby(['key1', 'key2']).agg([sum, min, max])
 
     def time_different_python_functions_singlecol(self, df):
-        df.groupby('key1').agg([sum, min, max])
+        df.drop(['key2'], axis=1).groupby('key1').agg([sum, min, max])
```
Comment (WillAyd) [inline asv_bench/benchmarks/groupby.py]: I think the problem here is that performance issues are going to be ambiguous between the drop and the groupby ops. Do you see a better way to refactor this test (maybe to a different class?) to get the intended effect?

Comment (qwhelan) [reply asv_bench/benchmarks/groupby.py]: Yeah, I'd just change `setup_cache()` to `setup()`, but that'd require slightly changing all the tests in this suite.

**Comment by jreback** [comment]
I believe this mostly been fixed in master

**Comment by qwhelan** [comment]
@WillAyd I don't believe that case is being hit here - the issue is that the multicolumn case aggregates for columns `value1`, `value2`, `value3`, which are all numeric (and thus fast due to cython path). The single column case also includes the column `key2`, which is a `str` column and thus must use the slower pure python path.

These factors lead to the misleading output of the single col case being 6x slower than the multicol case.

@jreback The typo was fixed in master, but there's an issue in the benchmark definition still.

**Comment by WillAyd** [comment]
My point is that the aggregations should not be applied to the object column. It only seems to do that when supplying an iterable of functions, whereas applying individually does the right thing

**Comment by qwhelan** [comment]
@WillAyd Okay, I can add separate benchmarks for the single function case but the intent of the existing benchmark seems to be to compare multiple aggregation performance with multiple aggregations.

**Comment by jreback** [comment]
@qwhelan is this still relevant?

**Pull request closed without merging**